src/components/LibraryItem.tsx

```diff
@@ -184,9 +185,6 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
     }
 
     onLibraryItemClicked() {
-
-        // Toggle expansion state.
-        let currentlyExpanded = this.state.expanded;
-        this.setState({ expanded: !currentlyExpanded });
+        LibraryEntryPoint.dispatchEvent("libraryItemClick", this);
```

Comment:
Why are we moving the stuff that is for `LibraryItem` out of it? This is what is expected:

```ts
onLibraryItemClicked() {
    let currentlyExpanded = this.state.expanded;
    this.setState({ expanded: !currentlyExpanded });

    // I think we need a global instance of Reactor object
    reactor.raiseEvent("itemClicked", this. data.creationName);
}

```